oxidation/libparsec/crates/platform_http_proxy/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Default)]
+pub struct ProxyConfig {
+    http_proxy: Option<String>,
+    https_proxy: Option<String>,
+}
+
+impl ProxyConfig {
+    /// Will configure the proxy using the environment variables:
+    ///
+    /// - `HTTP_PROXY` env var. will be used to configure the http proxy.
+    /// - `HTTPS_PROXY` env var. will be used to configure the https proxy.
+    ///
+    /// On `wasm32` this function will do nothing has we cant have access to the env variables since we will be run in a web navigator context.
+    pub fn with_env(self) -> Self {
+        self.with_env_internal()
+    }
+
+    /// Set the http proxy to use, will overwrite the previous configuration.
+    pub fn with_http_proxy<S: AsRef<str>>(mut self, proxy: S) -> Self {
+        self.http_proxy.replace(proxy.as_ref().to_string());
+        self
+    }
+
+    /// Set the https proxy to use, will overwrite the previous configuration.
+    pub fn with_https_proxy<S: AsRef<str>>(mut self, proxy: S) -> Self {
+        self.https_proxy.replace(proxy.as_ref().to_string());
+        self
+    }
+}
+
+impl ProxyConfig {
+    /// Configure the http client builder with the proxy configuration,
+    ///
+    /// On `wasm32` this will do nothing has `wasm32` `reqwest` don't expose `Proxy`
+    /// And we likely prefer use the web navigator to handle the proxy configuration.
+    pub fn configure_http_client(
+        &self,
+        builder: reqwest::ClientBuilder,
+    ) -> reqwest::Result<reqwest::ClientBuilder> {
+        self.configure_http_client_internal(builder)
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+mod not_wasm32 {
+    use std::{iter::Chain, option::IntoIter};
+
+    use reqwest::Proxy;
+
+    use super::ProxyConfig;
+
+    impl ProxyConfig {
+        pub(crate) fn with_env_internal(self) -> Self {
+            let cls = if let Ok(proxy) = std::env::var("HTTP_PROXY") {
```

Comment:
Use constant for `HTTP{,S}_PROXY` and modify them on test to be `TEST_HTTP{,S}_PROXY`